### PR TITLE
Adding Flake8 and Black formatting (take 2)

### PR DESCRIPTION
**Pull request by bernardopacini**
## Purpose
Adding Flake8 and Black formatting to all Python files in the repo (mdolab/private-docs#191). This includes the `examples/`, `pyspline/`, `src/f2py/`, `tests/reg_tests/` directories and package `setup.py` as well as docs `conf.py` files.

Black and Flake8 disagree on how to format the `setup.py` file and here we stick with Black formatting.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
No additional tests besides the already defined regression tests.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,24 +31,14 @@
...
-        "Programming Language :: Python, Fortran"]
-      )
-
+    long_description_content_type="text/markdown",
+    keywords="spline b-spline optimization",
+    author="",
+    author_email="",
+    url="https://github.com/mdolab/pyspline",
+    license="Apache License Version 2.0",
+    packages=["pyspline",],
```
Comment (nwu63) [inline setup.py:40]: The reason they disagreed has to do with the fact that there is a trailing comma. if you delete it then they should agree.

File: setup.py [hunk 2]
```diff
@@ -35,24 +31,14 @@
...
+    package_data={"pyspline": ["*.so"]},
+    install_requires=["numpy>=1.16", "scipy>=1.2",],
```
Comment (nwu63) [inline setup.py:42]: Same here.

**Review by nwu63 — CHANGES_REQUESTED** [review]
Just a few minor things.
- Flake8 still complains of unused imports in the importTest file (check the log). There is also a minor change in the `setup.py` file which I commented on below
- Black is still complaining. I suspect this is due to a different version of `black`. If you update yours to the latest version and run it again, everything should be fixed. But fix the flake8 stuff first

**Comment by bernardopacini** [comment]
For the first bullet, the import test only checks if the module can be imported, so it is entirely unused after being imported (intentionally). If I remove the import (as requested by Flake8) then this script would be useless and not actually check anything. I may be misunderstanding the purpose of this file so please correct me if I'm wrong. I'll go ahead and fix the `setup.py`, thank you for the heads up with that change.

For the second bullet, I double checked my black installation and it is up to date. This discrepancy may be caused by the fact that I'm running Python 3.9 on my machine, so I'll change that and make sure I'm checking for 3.7 - 3.8 instead (as Travis does).

**Comment by nwu63** [comment]
- Sorry should've been more clear. The way to address this is to add an inline comment `# noqa` which tells flake8 to ignore this error.
- As for black , I just tested this in a Python 3.9 environment and still getting `5 files reformatted` so something is going on. Maybe we can debug this offline.

**Comment by bernardopacini** [comment]
- Got it, I missed that feature so I'll update that. Thank you for the clarification

- Very strange. I'll take a look and see why I am getting the wrong behavior on my machine. Thanks for double checking on yours, sorry for the added hassle!

**Review by nwu63 — APPROVED** [review]

**Review by joanibal — APPROVED** [review]

**Pull request merged**